rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -534,6 +534,150 @@ void addSignatureMultipleInputsPartiallyValid() throws Exception {
+        // Act
+        List<BtcECKey> federatorPrivateKeys = BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS;
+        List<BtcECKey> privateKeysToSignWith = Arrays.asList(federatorPrivateKeys.get(0), federatorPrivateKeys.get(1));
+
+        // Federation is the genesis federation ATM
+        Federation federation = bridgeConstantsRegtest.getGenesisFederation();
+
+        final Keccak256 rskTxHash = PegTestUtils.createHash3();
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        BtcTransaction prevTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput prevOut = new TransactionOutput(btcRegTestParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        BtcTransaction btcTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput output = new TransactionOutput(btcRegTestParams, btcTx, Coin.COIN, new BtcECKey().toAddress(btcRegTestParams));
+        btcTx.addOutput(output);
+        btcTx.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+
+        Sha256Hash unsignedBtxTxHash = btcTx.getHash();
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP298)).thenReturn(true);
+
+        SortedMap<Keccak256, BtcTransaction> rskTxWaitingForSignatures = new TreeMap<>();
+        rskTxWaitingForSignatures.put(rskTxHash, btcTx);
+        when(provider.getRskTxsWaitingForSignatures()).thenReturn(
+            rskTxWaitingForSignatures
+        );
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstantsRegtest)
+            .withProvider(provider)
+            .withActivations(activations)
+            .build();
+
+        // Act
+        BtcECKey privateKeyToSignWith = privateKeysToSignWith.get(0);
+        List derEncodedSigs = getSignatureForTxInput(btcTx.getInputs().get(0), btcTx, privateKeyToSignWith);
+        bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeyToSignWith), derEncodedSigs, rskTxHash.getBytes());
+
+        // Verify pegout creation entry is only set to null when pegout is fully signed
+        verify(provider, never()).getPegoutCreationRskTxHashByBtcTxHash(any());
+        verify(provider, never()).setPegoutCreationEntry(
+            any()
+        );
+
+        BtcECKey privateKeyToSignWith2 = privateKeysToSignWith.get(1);
+        List derEncodedSigs2 = getSignatureForTxInput(btcTx.getInputs().get(0), btcTx, privateKeyToSignWith2);
+        bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeyToSignWith2), derEncodedSigs2, rskTxHash.getBytes());
+
+        // Assert
+        verify(provider, times(1)).getPegoutCreationRskTxHashByBtcTxHash(unsignedBtxTxHash);
+        verify(provider, never()).setPegoutCreationEntry(
```

Comment:
The test looks ok but I'm missing something. Where did you set it up so that it would not find the entry in the pegout creation index?